dev/check_pr_title.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
```

Comment:
```suggestion
#!/usr/bin/env python
# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
```

Should we make this dev script useable in the same way all other dev scripts are used?

Adding the shebang and making the script executable would turn

```
python ./dev/check_pr_title.py "${{ github.event.pull_request.title }}"
```
into
```
./dev/check_pr_title.py "${{ github.event.pull_request.title }}"
```

Reply:
Ok for me to do it or not. Independent of this, let's merge the PR asap.

Reply:
I think we can ignore this for now, as it only makes sense if we have no dependencies associated with the script. I think we can merge the PR

Reply:
@tanertopal btw, it won't be necessary for anyone to run this manually

Reply:
Got it. Let's merge it then and improve later. I personally would like to remove the YAML dependency and replace it with TOML (therefore use Py3.11 for this job) but all that can be done afterwards so this is already applied as you two suggested.